libsodium-bindings/src/LibSodium/Bindings/SealedBoxes.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE Trustworthy #-}
+
+-- |
+-- Module: LibSodium.Bindings.SealedBoxes
+-- Description: Direct bindings to the sealed boxes API of Libsodium
+-- License: BSD-3-Clause
+-- Maintainer: The Haskell Cryptography Group
+-- Stability: Stable
+-- Portability: GHC only
+module LibSodium.Bindings.SealedBoxes
+  ( -- * Introduction
+    -- $introduction
+
+    -- * Functions
+    cryptoBoxSeal
+  , cryptoBoxSealOpen
+
+    -- * Constants
+  , cryptoBoxSealbytes
+  ) where
+
+import Foreign (Ptr)
+import Foreign.C (CUChar, CInt(..), CULLong(..), CSize(..))
+
+-- $introdution
+-- Sealed boxes are designed to anonymously send messages to a recipient
+-- given their public key.
+--
+-- Only the recipient can decrypt these messages using their private key.
+-- While the recipient can verify the integrity of the message, they cannot
+-- verify the identity of the sender.
+--
+-- A message is encrypted using an ephemeral key pair, with the secret key being
+-- erased right after the encryption process.
+--
+-- Without knowing the secret key used for a given message, the sender cannot decrypt
+-- the message later. Furthermore, without additional data, a message cannot be
+-- correlated with the identity of its sender.
+
+-- | `cryptoBoxSeal` creates a new key pair for each message and attaches the public
+--   key to the ciphertext. The secret key is overwritten and is not accessible
+--   after this function returns.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_box_seal"
+  cryptoBoxSeal
+    :: Ptr CUChar
+    -- ^ Buffer that will hold the encrypted message of size
+    --   (size of original message + cryptoBoxSealbytes) bytes
+    -> Ptr CUChar
+    -- ^ Buffer that holds the plaintext message
+    -> CULLong
+    -- ^ Length of the plaintext message
+    -> Ptr CUChar
+    -- ^ Buffer that holds public key of size 'cryptoBoxPublicKeyBytes' bytes.
+    -> IO CInt
+    -- ^ Returns 0 on success and -1 on error.
+
+-- | `cryptoBoxSealOpen` doesn't require passing the public key of
+--   the sender as the ciphertext already includes this information.
+--
+--   Key pairs are compatible with operations from LibSodium.Bindings.CryptoBox
+--   module and can be created using cryptoBoxKeyPair or cryptoBoxSeedKeyPair.
```

Comment:
Could you perhaps use the single quotes syntax to create links? :)

Reply:
done